### DISCUSSION TITLE
Improve the semantics of `--after`/`--before`

### DISCUSSION THREAD

**Discussion by jotaen** [jotaen/klog#384] [Feature ideas & discussions]
This issue is a spin-off of https://github.com/jotaen/klog/issues/27. `--after` and `--before` are currently inclusive, which is not intuitive. On the other hand, there could also be legitimiate use cases for it, e.g. `--after=2000-01-01`, instead of `--after=1999-12-31`. I think the ultimate solution is this:

- [ ] Make `--after` and `--before` exclusive
- [ ] Offer a separate *inclusive* version of the flag ~e.g. `--after-or-at` or `--after-incl`~ `--since` and `--until`
- [ ] Support date ranges for convenience, i.e. `--in=2018` for “all in 2018” or `--in=2018-01` for “all in January 2018”.

All exact names tbd., suggestions welcome.

**Comment by vladdeSV** [comment]
Just a suggestion:
- `--since`
- `--until`

**Comment by wAuner** [comment]
@vladdeSV that's what I wanted to suggest as well. make after/before exclusive and since/until inclusive.
